From d485478a5b37663847037205005cf8e508402796 Mon Sep 17 00:00:00 2001
From: Jan Hambrecht <jaham@gmx.net>
Date: Sun, 14 May 2023 18:44:14 +0200
Subject: [PATCH] Video: Add bookmark state and watched progress bar

This adds bookmark state and a watched progress bar to the
video windows.
---
 base.xml     | 14 ++++++++++++++
 video-ui.xml | 21 +++++++++++++++++++--
 2 files changed, 33 insertions(+), 2 deletions(-)

diff --git a/base.xml b/base.xml
index dc2d313..79eb2c9 100644
--- a/base.xml
+++ b/base.xml
@@ -926,6 +926,17 @@
         </state>
     </statetype>
 
+    <statetype name="basebookmarkstate">
+        <area>0,0,32,32</area>
+        <state name="yes">
+            <area>0,0,100%,100%</area>
+            <imagetype name="icon" from="baseicon">
+                <position>0,0</position>
+                <filename>images/icons/Bookmark.png</filename>
+            </imagetype>
+        </state>
+    </statetype>
+
     <statetype name="baseplayliststate">
         <area>0,0,32,32</area>
         <state name="yes">
@@ -1048,6 +1059,9 @@
             <area>50,100%-32,-200,32</area>
             <alpha>150</alpha>
         </textarea>
+        <statetype name="bookmarkstate" from="basebookmarkstate">
+            <position>100%-72,100%-122</position>
+        </statetype>
         <statetype name="watchedstate" from="basewatchedstate">
             <position>100%-32,100%-122</position>
         </statetype>
diff --git a/video-ui.xml b/video-ui.xml
index e4b305d..709c44c 100644
--- a/video-ui.xml
+++ b/video-ui.xml
@@ -54,6 +54,9 @@
                         <area>0,0,100%+1,100%+1</area>
                         <type>box</type>
                     </shape>
+                    <progressbar name="buttonprogress1" from="basewatchedprogress">
+                        <area>0,100%+7,100%,3</area>
+                    </progressbar>
                 </state>
             </statetype>
         </buttonlist>
@@ -121,6 +124,9 @@
         <statetype name="userratingstate" from="basestarrating">
             <position>35,495</position>
         </statetype>
+        <progressbar name="watchedprogressbar" from="basewatchedprogress">
+            <area>35, 545, 100%-35, 3</area>
+        </progressbar>
         <textarea name="title" from="basetitletext">
             <area>235, 490, -235, 40</area>
             <align>top,hcenter</align>
@@ -128,6 +134,9 @@
             <scroll direction="horizontal" />
             <template>%TITLE| %%~ |00x00| %%~ |SUBTITLE%</template>
         </textarea>
+        <statetype name="bookmarkstate" from="basebookmarkstate">
+            <position>100%-107,495</position>
+        </statetype>
         <statetype name="watchedstate" from="basewatchedstate">
             <position>100%-67,495</position>
         </statetype>
@@ -251,6 +260,7 @@
                     <statetype name="watchedstate">
                         <alpha>255</alpha>
                     </statetype>
+                    <progressbar name="buttonprogress1" from="basewatchedprogress"/>
                 </state>
             </statetype>
             </buttonlist>
@@ -393,12 +403,12 @@
                         <alpha>150</alpha>
                     </textarea>
                     <textarea name="length" from="basetext_s">
-                        <area>100%-150,120,-10,30</area>
+                        <area>100%-160,120,-10,30</area>
                         <align>right,vcenter</align>
                         <alpha>150</alpha>
                     </textarea>
                     <statetype name="watchedstate" from="basewatchedstate">
-                        <position>100%-35,15</position>
+                        <position>100%-35,10</position>
                         <alpha>50</alpha>
                     </statetype>
                 </state>
@@ -428,6 +438,13 @@
                     <statetype name="watchedstate">
                         <alpha>255</alpha>
                     </statetype>
+                    <statetype name="bookmarkstate" from="basebookmarkstate">
+                        <position>100%-35,45</position>
+                        <alpha>255</alpha>
+                    </statetype>
+                    <progressbar name="buttonprogress1" from="basewatchedprogress">
+                        <area>125,100%-8,100%-5,3</area>
+                    </progressbar>
                 </state>
             </statetype>
         </buttonlist>
